### ISSUE TITLE
How to tune Gaussian Processes

### ISSUE THREAD

**Issue by Arktius** [fmfn/BayesianOptimization#202]
Hej folks,

I've read the [advanced-tour](https://github.com/fmfn/BayesianOptimization/blob/master/examples/advanced-tour.ipynb) but there's hardly any details regarding optimizing the GPs. Ok, there are the parameters n_restarts_optimizer, alpha, normalize_y and the kernel, but how do I tweak them or should I keep them as they are? Can I figure out, that one of these parameters isn't correctly set? Any advice someone?

**Comment by ben-arnao** [comment]
I am wondering the same thing. I know the Matern(nu=2.5) kernel seems to be pretty popular choice. I am curious if there are any guidelines for choosing alpha and the amount of restarts.

**Comment by samFarrellDay** [comment]
Matern is a popular choice because [this paper](https://papers.nips.cc/paper/4522-practical-bayesian-optimization-of-machine-learning-algorithms.pdf) provided experimental evidence that tended to cause faster convergence for hyperparameter searches.

Alpha in this package is passed to alpha in GaussianProcessRegressor, and represent's what's called the nugget factor. This factor tells the process that if you were to sample points at the same value, you would not always get the same result (i.e. x isn't perfectly correlated with x, so the diagonal of the covariance matrix should not be 1). I don't see where the documentation [here](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html) specifies if alpha is estimated, however the nugget factor and length scales are typically estimated using MLE or penalized MLE, so for the purpose of Bayesian Optimization you probably don't need to worry about tuning them.

n_restarts_optimizer simply refers to how many optimization attempts are made, the length scales that result in the minimum log-likelihood are chosen. Training Gaussian Processes is cheap if you have a low number of dimensions and samples, so I would set this high if your function is very expensive or high dimensional, and you want to be sure your GP doesn't get caught in a local minimum. There is no optimal value for this - higher values take more time but give you a higher chance to find the optimal parameters for the GP.

normalize_y simply scales the response - this is useful if your response has very high or low possible values - this is a similar concept to scaling your values for a neural network.

Overall I am not familiar with any studies done on performing parameter tuning for GP mid-optimization. Intuitively, I don't think it would give you much benefit, but maybe I am wrong.

**Comment by bwheelz36** [comment]
This is a really helpful answer. I would also like to suggest anyone reading this also reads #287, which discusses the impact of setting/not setting length_scale in the Kernel
